src/actions/actions.ts

```diff
@@ -590,10 +590,17 @@ class CommandInsertInInsertMode extends BaseCommand {
         vimState.cursorStartPosition = leftPosition;
       }
     } else {
-      await TextEditor.insert(char, vimState.cursorPosition);
+      const selection = vscode.window.activeTextEditor.selection;
+      if (selection.isEmpty) {
+        await TextEditor.insert(char, vimState.cursorPosition);
+      } else {
+        await TextEditor.replace(selection, char);
+      }
+
+      const cursorPosition = vscode.window.activeTextEditor.selection.end;
 
-      vimState.cursorStartPosition = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.start);
-      vimState.cursorPosition = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.start);
+      vimState.cursorStartPosition = Position.FromVSCodePosition(cursorPosition);
```

Comment:
My bad. I got confused because generally in VSCodeVim we use cursorPosition
to indicate our own Position variable, so I read the code incorrectly.

On Tue, Aug 2, 2016 at 3:04 PM, Pascal Hertleif <notifications@github.com>
wrote:

> In src/actions/actions.ts
> <https://github.com/VSCodeVim/Vim/pull/544#discussion_r73172814>:
>
> >
> > -      vimState.cursorStartPosition = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.start);
> > -      vimState.cursorPosition = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.start);
> > +      vimState.cursorStartPosition = Position.FromVSCodePosition(cursorPosition);
>
> But why would you even need to convert cursorPostition from a vscode
> position in the first place - it should already be the right type of
> position.
>
> I didn't dig any deep than:
>
> Type 'vscode.Position' is not assignable to type 'Position'.
>   Property '_nonWordCharRegex' is missing in type 'Position'.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/VSCodeVim/Vim/pull/544/files/e2aa40a22dca3e8d515c0e4628aa5327fa5984da#r73172814>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKPQSPGLAQfqpvmzUpONXD8hjPb6sr1ks5qb1yHgaJpZM4JZrc->
> .
>



-- 
Grant